### PR TITLE
Round on arm

### PR DESCRIPTION
**Pull request by mshabunin**
Added ARM implementation in cvRound function
    
    - note: uses VFPv3 instructions
    - also added overloaded cvRound variants with float and int parameters
    - thanks to Marina Kolpakova from Itseez for idea
    - thanks to developers from #llvm IRC channel for help with inline asm

### CODE REVIEW

**Comment by vpisarev** [comment]
@mshabunin, when PR passes all the builders, may I ask you to do 2 more things?
1) clean up the patch, remove commented-off code.
2) add credits somewhere (comments, commit message etc.) to Marina (unless she explicitly told you not to mention her)

File: modules/core/include/opencv2/core/cvdef.h [hunk 1]
```diff
@@ -455,6 +467,13 @@ CV_INLINE int cvRound( double value )
         fistp t;
     }
     return t;
+#elif (defined __GNUC__ && defined __arm__ && (defined __ARM_PCS_VFP || defined __ARM_VFPV3__))
```
Comment (ilya-lavrenov) [inline modules/core/include/opencv2/core/cvdef.h]: please, insert this code after Tegra one.

**Comment by mshabunin** [comment]
@vpisarev, the PR is ready. I have also fixed some warnings in iOS 64-bit builds.

**Comment by vpisarev** [comment]
great job! :+1:

**Pull request merged**